### PR TITLE
fix bad parsing of duration for Vue.js projects

### PR DESCRIPTION
**Pull request by ThomasCarca**
Fix [#12465](https://github.com/jhipster/generator-jhipster/issues/12465)

### CODE REVIEW

**Comment by ThomasCarca** [comment]
I was inspired by [this file](https://github.com/jhipster/generator-jhipster/blob/master/generators/client/templates/angular/src/main/webapp/app/shared/duration.pipe.ts.ejs) which works fine for Angular.

I just replicated the same logic to the util that parse duration for Vue; There should be, however, a way to mutualize this kind of code so we parse Durations the same way between frameworks

File: generators/client/templates/vue/src/main/webapp/app/shared/date/filters.ts.ejs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import Vue from 'vue';
-import { format, parseISO, formatDuration } from 'date-fns';
-import * as duration from 'duration-fns';
+import { format, parseISO } from 'date-fns';
+import * as moment from 'moment';
```
Comment (MathieuAA) [inline generators/client/templates/vue/src/main/webapp/app/shared/date/filters.ts.ejs:3]: Why do we use both libs? One lib isn't enough?

Comment (MathieuAA) [reply generators/client/templates/vue/src/main/webapp/app/shared/date/filters.ts.ejs:3]: I'm creating an issue for that, it's not related to the PR

Comment (ThomasCarca) [reply generators/client/templates/vue/src/main/webapp/app/shared/date/filters.ts.ejs:3]: We previously used only date-fns in this util but to be inline with what is done for Angular, we can use moment (see my comment on the PR)

I think date-fns could be entirely replaced by moment but did not wanted to go through this process as it is not in the scope of this PR

Comment (MathieuAA) [reply generators/client/templates/vue/src/main/webapp/app/shared/date/filters.ts.ejs:3]: Thanks for the explanation @ThomasCarca :)

**Comment by pascalgrimaud** [comment]
Yes, moment should not be used. I'm not sure we have the dependency either

**Comment by ThomasCarca** [comment]
> Yes, moment should not be used. I'm not sure we have the dependency either

It is used for Angular. Should we use different time libraries between different frameworks ?

**Comment by atomfrede** [comment]
We have not used moment for vue to have a very small bundle size. Of course as it is now part of core we could revisited the decision but I would not integrate moment.

**Comment by deepu105** [comment]
Yes, we use moment very littel, ideally we should migrate to a smaller lib if possible for all frameworks

**Comment by DanielFran** [comment]
Superseded by #12727

**Pull request closed without merging**